### ISSUE TITLE
A11y: Modal aria labels

### ISSUE THREAD

**Issue by silvalaura** [cengage/react-magma#1315]
We can improve the accessibility in our modals by updating the following:

- `aria-describedBy=“modal”` — This is not required and should be removed
- `aria-label=“modal”` — We hard code this right now and it’s fine that it’s hard coded as long as a user is using the heading. This is because `aria-labeledby` takes precedence over `aria-label`. For modals that don't have a `header` prop, we need to allow them to pass in a custom label. We should add a `ariaLabel` prop and do something like `aria-label={header ? null : props.ariaLabel}`

**Comment by orion-cengage** [comment]
I reviewed the the modal update in Dev Storybook and it looks good to me.

**Comment by ccedrone** [comment]
In Storybook > Drawer > Site Navigation, I don't see an `aria-label` on the Drawer, everything else looks good though.
